exporter/sumologicexporter/config.go

```diff
@@ -73,13 +74,8 @@ func createDefaultClientConfig() confighttp.ClientConfig {
 
 func (cfg *Config) Validate() error {
 
-	switch cfg.CompressEncoding {
-	case configcompression.TypeGzip:
-	case configcompression.TypeDeflate:
-	case NoCompression:
-
-	default:
-		return fmt.Errorf("invalid compression encoding type: %v", cfg.CompressEncoding)
+	if cfg.CompressEncoding != NoCompression {
+		return fmt.Errorf("support for compress encoding type %v has been deprecated, please use compression instead", cfg.CompressEncoding)
```

Comment:
```suggestion
		if cfg.CompressEncoding != nil {
		return errors.New("support for `compress_encoding` configuration has been removeed in favor of `compression`")
```

Reply:
> I tested the binary, and it seems that those changes are required, otherwise setting empty `compress_encoding` (which is valid for deprecated value) do not raise an error

good catch, I assumed`fmt.Errorf` would've raise an error, I guess not 😆